scripts/blt/cache.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @file
+ * Creates a unique temporary cache directory given a site, env, and uri.
+ * This is used for every blt command invocation on site install and update.
+ */
+
+$site = $argv[1];
+$env = $argv[2];
+$uri = $argv[3];
+
+
+if (empty($argv[3])) {
+  echo "Error: Not enough arguments. Site, environment, and uri are required.\n";
+  exit(1);
+}
+
+// Create a temporary cache directory for this drush process only.
+$cache_directory = sprintf('/mnt/tmp/%s.%s/drush_tmp_cache/%s', $site, $env, md5($uri));
```

Comment:
How is this functionally different from the existing code? It looks like this PR is assigning a cache directory based on site URI, whereas the existing site is assigning it based on db_role. Shouldn't that basically do the same thing?

Reply:
passing the directory as a cli prefix ensures that the child processes spawned from the current shell inherit the value. Exporting the value in the bash shell could still result in another process changing or overriding what should be a unique value. The db_role alone isnt enough to isolate the calls to cache rebuild in particular because on ACSF the live and update environments share a database, so env differentiates the two. It is more significant that that the cache directly is in /mnt as opposed to /tmp and that this directory is also created and deleted to avoid permissions issues and race conditions

Reply:
My understanding is that using `export` makes the variable available to all child / sub-processes. Ref: https://stackoverflow.com/questions/1158091/defining-a-variable-with-or-without-export

Reply:
the issue is that the value can then be modified by exec or other subprocesses of the blt and respective drush commands which does not prevent cache collisions on environments with multiple web servers. if you have suggestions on how to better solve this I am happy to refactor.